src/wasm/asm-wasm-builder.cc

```diff
@@ -369,6 +378,18 @@ class AsmWasmBuilderImpl : public AstVisitor {
 
   void VisitAssignment(Assignment* expr) {
     DCHECK(in_function_);
+    if (expr->value()->IsBinaryOperation() &&
```

Comment:
Actually could you switch this to get out the AsBinaryOperation value once + check for null / DCHECK non-null and keep using ISBinaryOperation?